### PR TITLE
Feature2_add_reducers_and_actions

### PR DESCRIPTION
**Pull request by Goldinium**
- Added Redux Toolkit (`npm install react-redux @reduxjs/toolkit`).
- Created a directory that contain all the Redux logic (`/src/redux`)
- Configured a Redux store (`/src/redux/store.js`)
- Defined a `slice` of state for `books` that:
   - Included an **array of books** (initial state: empty array)
   - Included a **reducer that adds a book**
   - Included a **reducer that removes a book**
- Defined a `slice` of state for `categories` that:
   - Included an **array of categories** (initial state: empty array)
   - Included a **reducer that checks the status** and always returns "Under construction" (the initial state should check to that string)
- Structured the application files using a "feature folder" approach and used the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for the Redux files. the files/folder structure look like this:
   ```sh
   /src
       |---/components
       |---/redux
           |--- /books
                   | booksSlice.js
           |--- /categories
                   | categoriesSlice.js
           | store.js
       | index.js
   ```

Thank you!

### CODE REVIEW

File: src/redux/books/bookSlice.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  books: [],
+};
+
+export const bookSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state) => {
+      state.value -= 1;
+    },
```
Comment (levy002) [inline src/redux/books/bookSlice.js:13]: - [ ] You're trying to update a value property in your `addBook` reducer, but there is no `value` property in your initial state.  you're decreasing `state.value`, which doesn't seem related to books. Kindly fix it.

- [OPTIONAL] I highly recommend you add the unique identifier `id` property to each newly added book which will help you to manipulate your books easily.

```suggestion
  addBook: (state, action) => {
      state.books = [
        ...state.books,
        {
          id: Math.floor(Math.random() * 600),
          title: action.payload.title,
          author: action.payload.author,
        },
      ]
    }
```

File: src/redux/books/bookSlice.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    removeBook: (state, action) => {
+      state.value += action.payload;
+    },
```
Comment (levy002) [inline src/redux/books/bookSlice.js:16]: - [ ] You're trying to update a value property in your `addBook` reducer, but there is no `value` property in your initial state. You're increasing `state.value`, which might not be the intended behavior. If you want to remove a book, you should filter it out from the `state.books` array.


```suggestion
 removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
```

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @Goldinium   , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter error :heavy_check_mark: 

## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [OPTIONAL] Kindly make sure next time to write a meaningful commits message.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @Goldinium, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  Implemented required changes. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
